frontend/src/metabase-lib/v1/expressions/config.ts

```diff
@@ -64,6 +64,18 @@ export const MBQL_CLAUSES: MBQLClauseMap = {
     args: ["number"],
     requiresFeature: "standard-deviation-aggregations",
   },
+  offset: {
+    displayName: `Offset`,
+    type: "aggregation",
+    args: ["any", "number"],
+    requiresFeature: "window-functions",
+    validator: function (_expr: any, offset: number) {
+      if (offset === 0) {
```

Comment:
Well we could do that, my main reason for disallowing it was to make the behavior more obvious to people using it. If you try to set it use it with zero and it just does nothing different that's probably going to be more confusing for people than if we just tell right away that you're not allowed to set it to zero, because it's meaningless. Better to get that feedback instantly than after waiting 10 seconds to run a query

Reply:
It definitely makes sense for a _hardcoded_ 0.
Just to give more context: my imagination brought me to cases like `Offset(Something, x - y)` where allowing 0 could make sense. I don't have a real-life use case example for that but from _math_ point of view that would make the feature "complete".

Reply:
Doing math here isn't supported, I don't know if that even works or not but it's not something I'm planning on supporting for the v1 or v2 of this feature.